modules/transport/http/src/org/apache/axis2/transport/http/HTTPSender.java

```diff
@@ -323,45 +322,9 @@ private void handleResponse(MessageContext msgContext,
                 }
             }
         }
-        if (HTTPStatusCodeFamily.SUCCESSFUL.equals(family)) {
-            // Save the HttpMethod so that we can release the connection when cleaning up
-            // HTTP 202 Accepted should support body based on ESBJAVA-4370 as spec does not strictly enforce
-            // no-entity-body with 202 Accepted response.
+        if (!errorCodes.contains(statusCode)) {
             msgContext.setProperty(HTTPConstants.HTTP_METHOD, method);
             processResponse(method, msgContext);
-        } else if (!errorCodes.contains(statusCode)
-                && (statusCode == HttpStatus.SC_INTERNAL_SERVER_ERROR
-                        || statusCode == HttpStatus.SC_BAD_REQUEST || statusCode == HttpStatus.SC_CONFLICT)) {
-            // Save the HttpMethod so that we can release the connection when cleaning up
-            msgContext.setProperty(HTTPConstants.HTTP_METHOD, method);
-            Header contenttypeHeader =
-                    method.getResponseHeader(HTTPConstants.HEADER_CONTENT_TYPE);
-            String value = null;
-            if (contenttypeHeader != null) {
-                value = contenttypeHeader.getValue();
-            }
-             OperationContext opContext = msgContext.getOperationContext();
-            if(opContext!=null){
-                MessageContext inMessageContext =
-                        opContext.getMessageContext(WSDLConstants.MESSAGE_LABEL_IN_VALUE);
-                if(inMessageContext!=null){
-                    inMessageContext.setProcessingFault(true);
```

Comment:
Did we check the effect of removing this line ?

Reply:
I ran mediation profile test cases with the fix and made sure there were no test failures.